File: 682-baseball-game/baseball-game.js
/**
 * @param {string[]} operations
 * @return {number}
 */
var calPoints = function(operations) {
    let score = [];
    let final = 0;
    for (o of operations) {
        if (Number(o)) {
            score.push(Number(o));
            final += score[score.length -1];
        } else if (o === 'C') {
            final -= score.pop();
        } else if (o === '+') {
            score.push(score[score.length -1] + score[score.length -2])
            final += score[score.length -1];
        } else if (o === 'D') {
            score.push(score[score.length -1] * 2);
            final += score[score.length -1];
        }
    }
    
    return final; 
};

File: 682-baseball-game/baseball-game.py
class Solution:
    def calPoints(self, operations: List[str]) -> int:
        scores = []
        sum = 0
        for o in operations:
            try:
                scores.append(int(o))
                sum += int(o)
            except ValueError:
                pass
            if o == '+':
                scores.append(scores[-1] + scores[-2])
                sum += scores[-1]
            elif o == 'D':
                scores.append(scores[-1] * 2)
                sum += scores[-1]
            elif o == 'C':
                sum -= scores[-1]
                scores.pop()
        return sum